addons/account/models/account_move.py

```diff
@@ -1674,36 +1674,34 @@ def _check_balanced(self, container):
         # It happens as the ORM makes the create with the 'no_recompute' statement.
-        self.env['account.move.line'].flush_model(['balance', 'currency_id', 'move_id'])
+        self.env['account.move.line'].flush_model(['balance', 'credit', 'debit', 'currency_id', 'move_id'])
         self.env['account.move'].flush_model(['journal_id'])
         self._cr.execute('''
-            WITH error_moves AS (
-                SELECT line.move_id,
-                       ROUND(SUM(line.balance), currency.decimal_places) balance
-                  FROM account_move_line line
-                  JOIN account_move move ON move.id = line.move_id
-                  JOIN account_journal journal ON journal.id = move.journal_id
-                  JOIN res_company company ON company.id = journal.company_id
-                  JOIN res_currency currency ON currency.id = company.currency_id
-                 WHERE line.move_id IN %s
-              GROUP BY line.move_id, currency.decimal_places
-            )
-            SELECT *
-              FROM error_moves
-             WHERE balance !=0
+            SELECT line.move_id,
+                   ROUND(SUM(line.debit), currency.decimal_places) debit,
+                   ROUND(SUM(line.credit), currency.decimal_places) credit
+              FROM account_move_line line
+              JOIN account_move move ON move.id = line.move_id
+              JOIN account_journal journal ON journal.id = move.journal_id
+              JOIN res_company company ON company.id = journal.company_id
+              JOIN res_currency currency ON currency.id = company.currency_id
+             WHERE line.move_id IN %s
+          GROUP BY line.move_id, currency.decimal_places
+            HAVING ROUND(SUM(line.balance), currency.decimal_places) != 0   
         ''', [tuple(moves.ids)])
 
         query_res = self._cr.fetchall()
         if query_res:
-            error_msg = _("There was a problem with the following move(s):\n")
+            error_msg = _("An error has occured")
             for move in query_res:
-                id_, balance = move
-                error_msg += _("- Move with id %i\n", id_)
-                if balance != 0.0:
-                    error_msg += _("\tCannot create unbalanced journal entry. The balance is equal to %s\n",
-                                   format_amount(self.env, balance, self.env['account.move'].browse(id_).currency_id))
+                move_id, debit, credit = move
+                currency = self.env['account.move'].browse(move_id).currency_id
+                error_msg += _(", the move is not balanced.\nThe total of debits equals %s and the total of credits equals %s."
+                               "\nYou might want to specify a default account on the miscellaneous journal to automatically balance each move.",
+                               format_amount(self.env, debit, currency),
+                               format_amount(self.env, credit, currency))
```

Comment:
* Include format the string like it is rendered (regarding the `\n`)
* Give the (display) name of the move, it is not easy otherwise to know what failed when doing batch operations
* Your way of starting with a `,` didn't really render nicely when multiple moves were failing

```suggestion
                error_msg += _(
                    "\n\n",
                    "The move (%s) is not balanced.\n"
                    "The total of debits equals %s and the total of credits equals %s.\n"
                    "You might want to specify a default account on the miscellaneous journal to automatically balance each move.",
                    move.display_name,
                    format_amount(self.env, sum_debit, move.currency_id),
                    format_amount(self.env, sum_credit, move.currency_id),
                )
```